hook!-full-game/src/Player/Player.gd

```diff
@@ -24,6 +24,7 @@ func _ready() -> void:
 
 func take_damage(source: Hit) -> void:
 	stats.take_damage(source)
+	state_machine.transition_to("Stagger")
```

Comment:
Actually, you don't want to do that. You should react to a signal on stats, something like "damage_taken". So we don't introduce bugs if e.g. we add an invincibility buff, an armor spell, something that prevents from taking damage in the future.

I think that taking damage in the hook state shouldn't stagger the character, so I'd put the signal reaction inside the states as well. Mainly on Move for now. You can use the `Events` autoload to connect Move to Stats without having to pass one to the other for now.

Reply:
I used `_player.stats` instead